core/src/main/java/com/linecorp/armeria/server/healthcheck/ScheduledHealthChecker.java

```diff
@@ -0,0 +1,216 @@
+
+        impl.compareAndSet(current, null);
+        current.stopHealthChecker();
+        current.removeListener(onHealthCheckerUpdate);
+    }
+
+    /**
+     * Used for test verification.
+     */
+    @VisibleForTesting
+    int getRequestCount() {
+        return requestCount.get();
+    }
+
+    /**
+     * Used for test verification.
+     */
+    @VisibleForTesting
+    boolean isActive() {
+        return impl.get() != null;
+    }
+
+    /**
+     * Health checker can be scheduled only once. Calling startHealthChecker won't work after stopHealthChecker.
+     */
+    private static class ScheduledHealthCheckerImpl implements ListenableHealthChecker {
+
+        private final Supplier<? extends CompletionStage<HealthCheckStatus>> healthChecker;
+        private final Duration fallbackTtl;
+        private final SettableHealthChecker settableHealthChecker;
+        private final EventExecutor eventExecutor;
+
+        private State state;
+        private volatile Future<?> scheduledFuture;
+
+        ScheduledHealthCheckerImpl(Supplier<? extends CompletionStage<HealthCheckStatus>> healthChecker,
+                                   Duration fallbackTtl, EventExecutor eventExecutor) {
+            this.healthChecker = healthChecker;
+            this.fallbackTtl = fallbackTtl;
+            this.eventExecutor = eventExecutor;
+            settableHealthChecker = new SettableHealthChecker(false);
+
+            state = State.INIT;
+            scheduledFuture = Futures.immediateVoidFuture();
+        }
+
+        @Override
+        public boolean isHealthy() {
+            return settableHealthChecker.isHealthy();
+        }
+
+        @Override
+        public void addListener(Consumer<? super HealthChecker> listener) {
+            settableHealthChecker.addListener(listener);
+        }
+
+        @Override
+        public void removeListener(Consumer<?> listener) {
+            settableHealthChecker.removeListener(listener);
+        }
+
+        private void startHealthChecker() {
+            if (state != State.INIT) {
+                return;
+            }
+            state = State.SCHEDULED;
+            eventExecutor.execute(this::runHealthCheck);
+        }
+
+        private void stopHealthChecker() {
+            if (state != State.SCHEDULED) {
+                return;
+            }
+            state = State.FINISHED;
+            scheduledFuture.cancel(true);
+        }
+
+        private void runHealthCheck() {
+            if (state != State.SCHEDULED) {
+                return;
+            }
+            try {
+                healthChecker.get().handle((result, throwable) -> {
+                    final boolean isHealthy;
+                    final long intervalMills;
+                    if (throwable != null) {
+                        isHealthy = false;
+                        intervalMills = fallbackTtl.toMillis();
+                    } else {
+                        isHealthy = result.isHealthy();
```

Comment:
If `result` is null, shouldn't we `isHealthy` is false and leave a warning message?

Reply:
updated, nice catch..:bow: 
Curious should we update the javadoc that we expect healthchecker should not return null or throwing exception. I treat exception & null as unexpected and logging with warn level.

Reply:
> should we update the javadoc that we expect healthchecker should not return null or throwing exception. 

That sounds nice.